File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.c
int lengthOfLongestSubstring(char* s) {
    int arr[157] = {0};
    int count = 0;
    int max = 0;
    for(int j=0;s[j]!='\0';j++){
        for(int i=j;s[i]!='\0';i++){
            if(arr[(int)s[i]]==0){
                arr[(int)s[i]] = 1;
                count++;
            }
            else{
                max = (max<count)?count:max;
                memset(arr, 0, sizeof(arr));
                count=0;
                break;
            }
        }
    }
    max = (max<count)?count:max;
    return max;
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> set = new HashSet<>();
        int maxLength = 0;
        int left = 0;
        for(int right=0;right<s.length();right++){
            char ch = s.charAt(right);
            if(!set.contains(ch)){
                set.add(ch);
                maxLength = Math.max(maxLength, right-left+1);
            } else{
                while(left<=right){
                    char tempCh = s.charAt(left);
                    if(tempCh==ch){
                        left++;
                        break;
                    } else{
                        set.remove(tempCh);
                    }
                    left++;
                }
            }
        }
        return maxLength;
    }
}